### ISSUE TITLE
Upgrading an encrypted room resulted in bad room state cache in the new room

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#8136]
After upgrading an internal room, many people ended up with one of two (or one then the other) problems with the room:
* Sending messages were unencrypted (indication of the client not seeing the state event). Receiving encrypted messages was fine.
* The room name was "Someone and N others" instead of the actual name (indication of the state event going missing).

Some people had to clear cache & reload multiple times.

This is almost certainly a duplicate of some other issue, however it is being reported for tracking that it can easily happen upon room upgrade. The room being encrypted may have nothing to do with this.

**Comment by lampholder** [comment]
@turt2live is this a `redesign` issue?

**Comment by turt2live** [comment]
I've flagged it as one because people on experimental had problems. I have not verified what app or develop does, but would expect the same.

**Comment by turt2live** [comment]
I think this is a subset of https://github.com/vector-im/riot-web/issues/7843 however an extremely easy way to reproduce it. Using this as the canonical "I upgraded a room and now everything looks broken" bug.

**Comment by turt2live** [comment]
Other symptoms people report:
* Lack of room name after upgrade
* Lack of merging in the room list
* Lack of "go back" button to see old room
* Lack of tombstone

**Comment by turt2live** [comment]
I haven't really been able to reproduce this in a monitored setting, and decided to go looking at known rageshakes to see what comes up. General report is available here on the sister issue: https://github.com/vector-im/riot-web/issues/7843#issuecomment-475453181

**Comment by turt2live** [comment]
Right, it's been 2.5 days of trying to figure out how to cause this reliably enough to debug it with no luck. If someone is unlucky enough to encounter this or #7843, please rageshake.
